policy/composer.go

```diff
@@ -90,10 +97,19 @@ func optimizeRule(ctx *cel.OptimizerContext, r *CompiledRule) (ast.Expr, bool) {
 			ctx.ReportErrorAtID(nestedRule.ID(), "subrule early terminates policy")
```

Comment:
We can effectively tip-toe around the subrule termination guard here because we're only checking for the optional properties of the immediate ancestor:

```
name: "subrule optional"
rule:
  match:
    - rule:
        match:
          - rule:
              match:
                - condition: >
                    x > 100
                  output: "'hi'"
          - output: "'this is effectively the default'"
    - output: "'we can never reach this'"
```

I think this problem likely existed before this fix, but probably good to take in consideration with https://github.com/google/cel-go/issues/985 if we end up supporting conditional nested rules.

Reply:
Definitely, the issue is mostly related to a lack of a condition expression when entering a rule which contains an `output` expression. I'll update this code path to reflect an `unreachable output` error in a subsequent PR for #985. It should be possible to detect through just a static pass over the outputs to see if there's more than one without any condition attached (directly, or in ancestor)